christin/index.js

```diff
@@ -0,0 +1,254 @@
+
+/* 11. XO Create a function that takes a string, checks if it has the same number of 'x's and 'o's and returns either true or false.
+
+Notes:
+Return a boolean value (true or false).
+The string can contain any character.
+When neither an x nor an o is in the string, return true.
+
+Examples:
+isEqualNumXandO("ooxx") ➞ true
+isEqualNumXandO("xooxx") ➞ false
+isEqualNumXandO("ooxXm") ➞ true (case insensitive)
+isEqualNumXandO("zpzpzpp") ➞ true (returns true if no x and o)
+isEqualNumXandO("zzoo") ➞ false*/
+
+function xO(string4) {
+  let x = 0;
+  let o = 0;
+  for (let i = 0; i < string4.length; i++) {
+    if (string4[i] === "x") {
+      x++;
+    } else if (string4[i] === "o") {
+      o++;
+    } else {
+      return "none of these";
+    }
+  }
+  let equal = x === o ? true : false;
+  return equal;
+}
+
+console.log("XO:", xO("xooxoxoxxox"));
+
+/* 12. isPrime? Create a function that returns true if a number is prime and false if it's not. NB: a prime number is any positive integer greater than 1, which is only evenly divisible by two divisors: itself and 1. The first ten prime numbers are 2, 3, 5, 7, 11, 13, 17, 19, 23 and 29.
+
+Examples:
+isPrime(7) ➞ true
+isPrime(9) ➞ false
+isPrime(10) ➞ false*/
+
+function findFactors(number4) {
+  res = [];
+  for (i = 1; i <= number4; i++) {
+    if (number4 % i === 0) {
+      res[res.length] = i;
+    }
+  }
+  return res;
+}
+
+function isPrime(number4) {
+  let factors = findFactors(number4);
+  if (factors.length === 2) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+console.log(isPrime(21));
+
+/* 13. Validate Email. Create a function that takes a string, checks for valid email address syntax, and then returns either true or false accordingly.
+
+NOTE: email syntax is more complicated than you might think (see https://en.wikipedia.org/wiki/Email_address#Syntax), so we do not expect you to cover all cases! Just work your way through the below simplified validation criteria:
+
+The string must contain exactly one "@" character.
+The string must contain at least one "." character.
+The "." and the "@" must be in the appropriate places.
+e.g. "." cannot be the last character.
+e.g. "." cannot be directly before or after "@"
+e.g. there cannot be consecutive "." characters
+e.g. "@" must have at least one character in front of it.
+e.g. "j@example.com" is valid while "@example.com" is invalid.
+e.g. "john.smith@com" is invalid while "john.smith@email.com" is valid.
+e.g. "john..smith@email.com" and "john.@email.com" and "john@.email.com" are all invalid.*/
+
+function isValidEmail(email) {
+  if (
+    email.includes(".@") ||
+    email.includes("@.") ||
+    email.includes("..") ||
+    email.endsWith("@") ||
+    email.endsWith(".") ||
+    email.startsWith("@")
+  ) {
+    return false;
+  } else if (email.includes("@") && email.includes(".")) {
+    return true;
+  } else {
+    return false;
+  }
+}
+console.log("valid mail", isValidEmail("@mail@gnk.com"));
```

Comment:
Task 13 ==>  "john.smith@com" must be invalid(false) but you get true!

Reply:
changes it, but its not the best way - could you expain please?

Reply:
// Good job, also please take a look at this solution, here we already defined  all true and false cases ==>
function validateEmail(string) {
    if (
      !string.includes("@") ||
      !string.includes(".") ||
      string.includes(".@") ||
      string.includes("@.") ||
      string.includes("..") ||
      string[string.length - 1] === "." ||
      string[0] === "@" ||
      !string.includes(".", string.indexOf("@"))
    ) {
      return false;
    } else {
      return true;
    }
  }
  console.log("Valid ==>",validateEmail("@e.com"));

Reply:
ah ok, that makes sense as well. short questions:
1.  string[string.length - 1] === "." || - means no '.' in the end right?
2. !string.includes(".", string.indexOf("@")) - what does that mean?

Reply:
1 ==> Indeed, it means dot is not allowed at the end
2   !string.includes(".", string.indexOf("@")) ==> If we delete this code and try to check if  john.smith@com is true or false, which must to be false, the result will be true. Means that our email address  should contain @ and somewhere after @ there should be a dot!

Reply:
AFTER the @ is defined because its written on second part? i dont understand the connection between the indexof"@" other than because its in second position.

sorry for all the questions

Reply:
No prob:)    you also could declare string.indexOf("@") first. Like ==>      !string.includes( string.indexOf("@"), ".")